src/Nethermind/Nethermind.Merge.Plugin/Handlers/GetPayloadV3Handler.cs

```diff
@@ -14,11 +15,17 @@ namespace Nethermind.Merge.Plugin.Handlers;
 /// </summary>
 public class GetPayloadV3Handler : GetPayloadHandlerBase<GetPayloadV3Result>
 {
-    public GetPayloadV3Handler(IPayloadPreparationService payloadPreparationService, ISpecProvider specProvider, ILogManager logManager) : base(
+    private readonly CensorshipDetector _censorshipDetector;
+    public GetPayloadV3Handler(
+        IPayloadPreparationService payloadPreparationService,
+        ISpecProvider specProvider,
+        ILogManager logManager,
+        CensorshipDetector censorshipDetector) : base(
```

Comment:
Make CensorhipDetector optional here, and ShouldOverrideBuilder should be false if it isn't provided.